### DISCUSSION TITLE
AstroNvim: When rustaceanvim is used instead of lspconfig, <leader>gd does not work

### DISCUSSION THREAD

**Discussion by FritsHoogland** [mrcjkb/rustaceanvim#158] [Q&A]
### Feature description

I am fairly new to using neovim/astrovim, so please bear with me.
With rust-analyzer installed, <leader>gd with a function under the cursor will jump to the definition. 
When the rustaceanvim is used, this key combination is "gone". In fact a lot of the functions that are in <leader>l with rust-analyzer are gone with rustaceanvim. 

The question is whether this is functionality that is provided by rust-analyzer, and not by rustaceanvim.

If it's specifically provided by rust-analyzer, and not by rustaceanvim, then it's obvious it's gone, and my request is whether this functionality can be provided with rustaceanvim too. It 's really convenient to move through a project going to the definitions with a key combination.

If it's generic functionality and not specifically provided by rust-analyzer, then the question is how to be able to use this functionality again.

**Comment by mrcjkb** [comment]
Hey :wave: 

Neither rustaceanvim, nor lspconfig.rust_analyzer create any keymaps.
It looks like astro-community is replacing rust-tools with rustaceanvim in their rust language pack: https://github.com/AstroNvim/astrocommunity/pull/631

So if you're using AstroNVim, maybe it's worth sticking with rust-tools and waiting for that PR to land.

For plain Neovim, the readme's [Quick Setup section](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#quick-setup) says how you can configure keymaps:

- In a `~/after/plugin/rust.lua` file.
- In the `vim.g.rustaceanvim.server.on_attach` function.
- With an `LspAttach` autocommand.

**Reply by FritsHoogland** [reply]
Thank you for your answer. 

The first question was whether this functionality was only provided by rust_analyzer, for which I've found that it doesn't: the functionality of "goto definition" is available with rustaceanvim too, which can be validated by running:
```
:lua vim.lsp.buf.definition
```
I've added a new keymap to my `after/ftplugin/rust.lua` file:
```
vim.keymap.set("n", "gd", function () vim.lsp.buf.definition() end, { buffer = bufnr, desc = "goto definition" })
```
Which now provides this functionality with "gd" again.
(sorry, the heading says '<leader>gd', which was not the correct combination, and is the 'git' command group)
